types/next/index.d.ts

```diff
@@ -24,15 +29,7 @@ declare namespace next {
         /** path section of URL */
         pathname: string;
         /** query string section of URL parsed as an object */
-        query: {
-            [key: string]:
-                | boolean
-                | boolean[]
-                | number
-                | number[]
-                | string
-                | string[];
-        };
+        query: QueryStringMapObject;
```

Comment:
You could probably replace `QueryStringMapObject` with `Record<string, string | string[] | undefined>` instead of using an index type (since you don't have any other fields).

But nbd, and probably just preference. :)

Reply:
@Dru89 Ah yes, good idea. Thanks!